cmake/modules/CheckCompiler.cmake

```diff
@@ -192,8 +193,19 @@ check_cxx_source_compiles(
 int main() {}
 " GLIBCXX_USE_CXX11_ABI)
 
+
 #---Print the final compiler flags--------------------------------------------------------------------
 message(STATUS "ROOT Platform: ${ROOT_PLATFORM}")
 message(STATUS "ROOT Architecture: ${ROOT_ARCHITECTURE}")
 message(STATUS "Build Type: ${CMAKE_BUILD_TYPE}")
-message(STATUS "Compiler Flags: ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${uppercase_CMAKE_BUILD_TYPE}}")
+
+string(TOUPPER ${CMAKE_BUILD_TYPE} upperCaseBuildType)
+message(STATUS "Compiler Flags (automatically from 'CMAKE_BUILD_TYPE'): ${CMAKE_CXX_FLAGS_${upperCaseBuildType}}")
+
+#---Fix order of compiler flags: user-supplied flags are overridden otherwise----------------------
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_${upperCaseBuildType}} ${CMAKE_CXX_FLAGS}")
```

Comment:
What if `CMAKE_CXX_FLAGS` already has `${CMAKE_CXX_FLAGS_${upperCaseBuildType}}` at the end? This will just add it also at the beginning, no?

Reply:
Can this happen? I'm not sure.
What's a bit nasty, though is that we should actually check if the build type is set at all if we remove the defaulting.